lepton: init at 1.2.1

artemist opened the pull request:
###### Motivation for this change
Lepton is Dropbox's JPEG compression program. It was not previously packaged.


###### Things done
Add a lepton package for Linux. Note that it requires some Linux-specific things (glibc.static) and I do not have a Mac to test the required modifications

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

artemist committed:
lepton: init at 1.2.1

Ma27 commented:
@GrahamcOfBorg build lepton

Ma27 commented:
Built and tested locally :+1: 

According to the ofBorg CI, it doesn't build on aarch64-linux as well. I'm currently trying to get the cross build running on my machine to see if we can do something about it.

Regarding the darwin issues: I'd leave it disabled for now until somebody with the needed resources can help out :)

artemist commented:
Even if we switch to using system dependencies (it wants a specific version of zlib and has it in-tree), I don't think it will work because it uses Intel SIMD intrinsics. There is SSE2NEON, but that may be outside of the scope of this package

Ma27 committed:
lepton: only build on x86_64-linux for now

Darwin and `aarch64-linux` builds currently fail[1][2].

Also cleaned up the `meta` section a bit.

[1] https://github.com/NixOS/nixpkgs/pull/56108/checks?check_run_id=66442317
[2] https://github.com/NixOS/nixpkgs/pull/56108#issue-254801596

Ma27 commented:
I tried a cross build with `nix-build -E "with import ./. { crossSystem.config = \"aarch64-linux\"; }; lepton"`, but I didn't even get the glibc to compile. I now pushed a fix which disables all builds except `x86_64-linux` for now, if builds on other platforms are needed and possible, we can still fix this afterwards.

If you're fine with my patch, I'd wait for ofBorg to complete and mergen then :)

artemist commented:
That looks good! I'm pretty new to Nix, so my code isn't the best

Ma27 commented:
@artemist thanks!
